kbase-extension/static/kbase/js/widgets/function_input/editors/kbaseMediaEditor.js

```diff
@@ -0,0 +1,504 @@
+/**
+ *	kbaseMediaEditor.js (kbaseMediaEditor)
+ *
+ *	Given a pair of workspace and media object names or ids,
+ *	produce an editable media table.
+ *
+ *  Notes:
+ *  	to be refactored to use an editor widget.
+ *
+ *  Authors:
+ * 		nconrad@mcs.anl.gov
+ *
+ */
+
+
+ define([
+    'jquery',
+    'plotly',
+    'ModelingAPI',
+    'kbwidget',
+    'kbaseAuthenticatedWidget',
+    'kbaseModal'
+],
+function($, Plotly, ModelingAPI) {
+
+
+'use strict';
+
+$.KBWidget({
+    name: "kbaseMediaEditor",
+    parent: "kbaseAuthenticatedWidget",
+    version: "1.0.0",
+    options: {},
+    init: function(input) {
+        this._super(input);
+        var self = this,
+            container = $('<div class="kb-editor">');
+
+        self.$elem.append(container);
+
+
+        var modeling = new ModelingAPI( self.authToken() ),
+            kbapi = modeling.api,
+            biochem = modeling.biochem,
+            getCpds = modeling.getCpds;
+
+        // accept either workspace/object names or workspace/object ids
+        var wsName = input.ws,
+            objName = input.obj;
+
+        if (isNaN(wsName) && isNaN(objName) )
+            var param = {workspace: wsName, name: objName};
+        else if (!isNaN(wsName) && !isNaN(objName) )
+            var param = {ref: wsName+'/'+objName};
+        else
+            console.error('kbaseMediaEditor arguements are invalid');
+
+        var table, // main table to be rendered
+            media, // actual media data
+            rawData;  // raw workspace object
+
+        // some controls for the table
+        var saveBtn = $('<button class="btn btn-primary edit-btn pull-right hide">'+
+                        'Save</button>');
+        var saveAsBtn = $('<button class="btn btn-primary edit-btn pull-right hide">'+
+                        'Save as...</button>');
+        var addBtn = $('<button class="btn btn-primary pull-right">'+
+                       '<i class="fa fa-plus"></i> Add compounds...</button>');
+        var rmBtn = $('<button class="btn btn-danger pull-right hide">');
+        var undoBtn = $('<button class="btn btn-danger edit-btn pull-right hide">'+
+                            '<i class="fa fa-undo"></i>'+
+                        '</button>');
+
+        // keep track of edits
+        var _editHistory = new EditHistory();
+
+        // get media data
+        container.loading();
+        kbapi('ws', 'get_objects', [param])
```

Comment:
The kbapi part - sorry, I thought you were assuming the older kbapi module was loaded. Didn't see the declaration above.